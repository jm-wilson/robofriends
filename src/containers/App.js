import React, { useEffect } from 'react';
import CardList from '../components/CardList.js';
import SearchBox from '../components/SearchBox.js';
import Scroll from '../components/Scroll.js';
import ErrorBoundry from '../components/ErrorBoundry.js';
import './App.css';

import { useSelector, useDispatch } from 'react-redux';
import { incrementClickme, requestRobots, filterRobots, setSearchField } from '../actions.js';

const App = () => {
  const counter = useSelector((state) => state.clickMe.counter);
  const robots = useSelector((state) => state.requestRobots.robots);
  const filteredRobots = useSelector((state) => state.filterRobots.robots);

  const dispatch = useDispatch();

  const onSearchChange = (event) => {
    dispatch(setSearchField(event.target.value));
    dispatch(filterRobots(event.target.value, robots));
  };

  const onClickMe = () => {
    dispatch(incrementClickme(counter));
  };

  useEffect(() => {
    requestRobots(dispatch);
  }, []);

  return (
    <div className='tc'>
      <h1 className='f1'>RoboFriends</h1>
      <button onClick={onClickMe}>Click me ({counter})</button>
      <SearchBox searchChange={onSearchChange} />
      <Scroll>
        <ErrorBoundry>
          <CardList robots={filteredRobots} />
        </ErrorBoundry>
      </Scroll>
    </div>
  );
};

export default App;
