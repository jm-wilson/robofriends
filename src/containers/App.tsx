import React, { useEffect } from 'react';
import CardList from '../components/CardList';
import CounterButton from '../components/CounterButton';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import ErrorBoundary from '../components/ErrorBoundary';
import './App.css';

import { useSelector, useDispatch } from 'react-redux';
import { requestRobots, filterRobots, setSearchField } from '../actions';

export interface IRobot {
  name: string;
  id: number;
  email: number;
}

export interface IRootState {
  requestRobots: {
    robots: IRobot[];
  };
  filterRobots: {
    robots: IRobot[];
  };
  clickMe: {
    counter: number;
  };
}

const App: React.FC = (): JSX.Element => {
  const robots = useSelector((state: IRootState): IRobot[] => state.requestRobots.robots);
  const filteredRobots = useSelector((state: IRootState): IRobot[] => state.filterRobots.robots);

  const dispatch = useDispatch();

  const onSearchChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
    dispatch(setSearchField(event.target.value));
    dispatch(filterRobots(event.target.value, robots));
  };

  useEffect(() => {
    dispatch(requestRobots());
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <div className='tc'>
      <h1 className='f1'>RoboFriends</h1>
      <CounterButton />
      <SearchBox searchChange={onSearchChange} />
      <Scroll>
        <ErrorBoundary>
          <CardList robots={filteredRobots} />
        </ErrorBoundary>
      </Scroll>
    </div>
  );
};

export default App;
