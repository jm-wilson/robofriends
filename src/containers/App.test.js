import { shallow } from 'enzyme';
import React from 'react';
import App from './App';

it('should render App component', () => {
  expect(shallow(<App />)).toMatchSnapshot();
});
