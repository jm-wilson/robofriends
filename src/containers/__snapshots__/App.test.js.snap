// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App component 1`] = `
<div
  className="tc"
>
  <h1
    className="f1"
  >
    RoboFriends
  </h1>
  <CounterButton />
  <SearchBox
    searchChange={[Function]}
  />
  <Scroll>
    <ErrorBoundry>
      <CardList />
    </ErrorBoundry>
  </Scroll>
</div>
`;
